tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+     );
+     $this->assertTrue(
+       method_exists($collection, 'setPluginId'),
+       'UserManagerInterface does not have fsetPluginId function/method'
+     );
+     $this->assertTrue(
+       method_exists($collection, 'getDrupalUserId'),
+       'UserManagerInterface does not have getDrupalUserId function/method'
+     );
+
+   }
+
+   /**
+    * Test for class UserManager
+    */
+
+   public function testUserManager () {
+     // assertion to check if the file exists
+     // $this->assertFileExists('../drupal8/modules/social_api/src/User/UserManager.php');
+
+     // creating a mock object and passing constructor paremeters
+     $this->entity_type_manager = $this->createMock(EntityTypeManagerInterface::class);
+     $this->messenger = $this->createMock(MessengerInterface::class);
+     $this->logger_factory = $this->createMock(LoggerChannelFactoryInterface::class);
+
+     $collection = $this->getMockBuilder('Drupal\social_api\User\UserManager')
+                        ->setConstructorArgs(array($this->entityType, $this->entity_type_manager, $this->messenger, $this->logger_factory))
+                        ->getMockForAbstractClass();
+     $this->assertTrue($collection instanceof UserManager);
+     // checking for correct setPluginId and getPluginId method
+     $collection->setPluginId('1234');
+     // var_dump($collection->getSessionPrefix());
+     $this->assertEquals('1234', $collection->getPluginId());
+   }
+
+   /**
+    * Test for class UserAuthenticator
+    */
+
+   public function testUserAuthenticator () {
+     // assertion to check if the file exists
+     // $this->assertFileExists('../drupal8/modules/social_api/src/User/UserAuthenticator.php');
+
+     // creating a mock object and passing constructor paremeters
+     $this->sessionKeys = array(1,2,3);
+     $this->current_user = $this->createMock(AccountProxyInterface::class);
+     $this->messenger = $this->createMock(MessengerInterface::class);
+     $this->logger_factory = $this->createMock(LoggerChannelFactoryInterface::class);
+     $this->user_manager = $this->createMock(UserManagerInterface::class);
+     $this->data_handler = $this->createMock(SocialApiDataHandler::class);
+
+     $collection = $this->getMockBuilder('Drupal\social_api\User\UserAuthenticator')
+                        ->setConstructorArgs(array($this->current_user, $this->messenger, $this->logger_factory, $this->user_manager, $this->data_handler))
+                        ->getMockForAbstractClass();
+     $collection->setPluginId('drupal123');
+     $this->assertEquals('drupal123', $collection->getPluginId());
+     $this->dataHandler = $this->data_handler;
+
+     $collection->setSessionKeysToNullify($this->sessionKeys);
+     if (!empty($this->sessionKeys)){
+       array_walk($this->sessionKeys, function ($session_key) {
+         $this->dataHandler->set($this->dataHandler->getSessionPrefix() . $session_key, NULL);
```

Comment:
You have tested absolutely nothing here. There are not assertions at all. You're also missing  `nullifySessionKeys`